Code/Framework/AzFramework/AzFramework/Components/NativeUISystemComponent.h

```diff
@@ -0,0 +1,67 @@
+#include <AzFramework/Input/Devices/Motion/InputDeviceMotion.h>
+#include <AzFramework/Input/Devices/Mouse/InputDeviceMouse.h>
+#include <AzFramework/Input/Devices/Touch/InputDeviceTouch.h>
+#include <AzFramework/Input/Devices/VirtualKeyboard/InputDeviceVirtualKeyboard.h>
+#include <AzFramework/Windowing/NativeWindow.h>
+
+namespace AzFramework
+{
+    //! This system will take care of reflecting physics material classes and
+    //! registering the physics material asset.
+    class NativeUISystemComponent
+        : public AZ::Component
+    {
+    public:
+        AZ_COMPONENT(AzFramework::NativeUISystemComponent, "{F2BA607F-08A9-42DD-9C0D-F324BFDD4DFD}");
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided);
+        static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible);
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+        static void GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent);
+
+        NativeUISystemComponent();
+        ~NativeUISystemComponent();
+
+        NativeUISystemComponent(const NativeUISystemComponent&) = delete;
+        NativeUISystemComponent& operator=(const NativeUISystemComponent&) = delete;
+
+        // AZ::Component overrides...
+        void Activate() override;
+        void Deactivate() override;
+
+    private:
+        Application::ImplementationFactory* CreateApplicationImplementationFactory() const;
+        InputDeviceGamepad::ImplementationFactory* GetDeviceGamepadImplentationFactory() const;
+        InputDeviceKeyboard::ImplementationFactory* GetDeviceKeyboardImplementationFactory() const;
+        InputDeviceMotion::ImplementationFactory* GetDeviceMotionImplentationFactory() const; 
+        InputDeviceMouse::ImplementationFactory* GetDeviceMouseImplentationFactory() const;
+        InputDeviceTouch::ImplementationFactory* GetDeviceTouchImplentationFactory() const;
+        InputDeviceVirtualKeyboard::ImplementationFactory* GetDeviceVirtualKeyboardImplentationFactory() const;
+        NativeWindow::ImplementationFactory* GetNativeWindowImplementationFactory() const;
+
+        Application::ImplementationFactory* m_applicationImplFactory = nullptr;
+        InputDeviceGamepad::ImplementationFactory* m_deviceGamepadImplFactory = nullptr;
+        InputDeviceKeyboard::ImplementationFactory* m_deviceKeyboardImplFactory = nullptr;
+        InputDeviceMotion::ImplementationFactory* m_deviceMotionImplFactory = nullptr;
+        InputDeviceMouse::ImplementationFactory* m_deviceMouseImplFactory = nullptr;
+        InputDeviceTouch::ImplementationFactory* m_deviceTouchImplFactory = nullptr;
+        InputDeviceVirtualKeyboard::ImplementationFactory* m_deviceVirtualKeyboardImplFactory = nullptr;
+        NativeWindow::ImplementationFactory* m_nativeWindowImplFactory = nullptr;
```

Comment:
These should be made `unique_ptr` per the [O3DE C++ Best Practices Guide](https://github.com/o3de/sig-core/blob/main/governance/C%2B%2B-Best-Practices-Guide.md#tips-when-to-use-specific-smart-pointer-types).

We shouldn't be using raw `delete` calls in the code possible unless when working with complex scenarios of mixed pointer ownership

```suggestion
        AZStd::unique_ptr<Application::ImplementationFactory> m_applicationImplFactory;
        AZStd::unique_ptr<InputDeviceGamepad::ImplementationFactory> m_deviceGamepadImplFactory;
        AZStd::unique_ptr<InputDeviceKeyboard::ImplementationFactory> m_deviceKeyboardImplFactory;
        AZStd::unique_ptr<InputDeviceMotion::ImplementationFactory> m_deviceMotionImplFactory;
        AZStd::unique_ptr<InputDeviceMouse::ImplementationFactory> m_deviceMouseImplFactory;
        AZStd::unique_ptr<InputDeviceTouch::ImplementationFactory> m_deviceTouchImplFactory;
        AZStd::unique_ptr<InputDeviceVirtualKeyboard::ImplementationFactory> m_deviceVirtualKeyboardImplFactory;
        AZStd::unique_ptr<NativeWindow::ImplementationFactory> m_nativeWindowImplFactory;
```